src/DepositHandler.sol

```diff
@@ -1,45 +1,58 @@
 // SPDX-License-Identifier: MIT
-pragma solidity 0.8.20;
+pragma solidity 0.8.28;
 
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/utils/Pausable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import {IDepositHandlerErrors} from "./interfaces/ICustomErrors.sol";
 
-contract DepositHandler {
+/**
+ * @title Deposit Handler contract.
+ * @author @ohMySol, @nynko, @ok567
+ * @dev Contract for managing users deposits for bootcamps.
+ * Implements both user part and admin part for deposit management.
+ * Apart from that contract allow to manage admins and bootcamps.
+ */
+contract DepositHandler is IDepositHandlerErrors, Pausable, Ownable {
+    IERC20 public immutable usdc; //at the moment assume that Encode using only USDC.
```

Comment:
It should indeed be immutable but in my opinion it should be better to make the name generic (despite thinking it may be most likely usdc). `erc20Token` may be better later. For now no need to change ! we can keep usdc until we find a generic solution for the decimals and so on...

Reply:
Yes, I also think so that it should be some generic naming, to allow them to create a contract for a bootcamp not only with USDC, but maybe with other Stables.